src/software/ai/passing/cost_function.cpp

```diff
@@ -13,7 +13,31 @@
 #include "software/geom/algorithms/convex_angle.h"
 #include "software/logger/logger.h"
 
-double ratePass(const World& world, const Pass& pass, const Rectangle& zone,
+double rateAnyPass(const World& world, const BasePass& pass, const Rectangle& zone,
+                TbotsProto::PassingConfig passing_config)
+{
+    if (pass.type() == PassType::CHIP_PASS)
+    {
+        std::vector<Robot> enemy_team_not_skipped;
+
+        for (auto &enemy_robot : world.enemyTeam().getAllRobots())
+        {
+            if (!reinterpret_cast<const ChipPass*>(&pass)->isSkipped(enemy_robot.position())) 
+            {
+                enemy_team_not_skipped.push_back(enemy_robot);
+            }
+        }
+        return ratePass(world, pass, zone, Team(enemy_team_not_skipped), passing_config);
+    }
+    else if (pass.type() == PassType::GROUND_PASS)
+    {
+        return ratePass(world, pass, zone, world.enemyTeam(), passing_config);
+    }
```

Comment:
maybe better for `BasePass` to have a function called `getIgnoredEnemyRobots(const World& world)`. I'd be happy to listen to any counter-arguments you have